integr8ly/integreatly-operator
# Creation of alerts for all RHMI service endpoints

## Description
https://issues.redhat.com/browse/INTLY-7403

The naming conventions for each alert along with the requirements for the alerts expressions can be found in [INTLY-7402](https://issues.redhat.com/browse/INTLY-7402).

## Verification Steps
- Install `RHMI` using this branch
- Ensure that none of the alerts are firing during install
- Ensure all of the required alerts are created. [This](https://docs.google.com/spreadsheets/d/1Wt6qrRskOQLd5DndMXvMvocsfsxcIOk7uk8TWsX4MBo/edit#gid=0) Google Sheet contains all of the alerts that were created (please note that there are 3 tabs)
- To test an individual alert, go to the product/operator namespace in which the alerts resides, and search for the `Endpoints` type resource (`home` -> `search` -> `Endpoints`). Within the YAML definitition of the resource, delete everything from the word `subsets` down
- After a couple of minutes, the corresponding alert should fire. Please note that the operator will eventually reconcile the `Endpoints` resource, so the alert will return to normal.
- Ensure that a SOP URL is also available for all of the newly created alerts. The SOP is still a WiP so the link will be dead until the SOP is completed and merged.

## Testing
The `alerts_exist.go` test file has been updated to include the alerts created in this PR.
- Install `RHMI` using this branch.
- Run the tests against your cluster using  `make test/functional`

## Type of change


## Checklist
